easybuild/framework/easyconfig/easyconfig.py

```diff
@@ -1305,6 +1327,15 @@ def _parse_dependency(self, dep, hidden=False, build_only=False):
         else:
             raise EasyBuildError("Dependency %s of unsupported type: %s", dep, type(dep))
 
+        if isinstance(dependency['version'], string_type):
+            self.log.debug("Dependency version is already a string ('%s'), OK", dependency['version'])
+        elif dependency['version'] is None:
```

Comment:
Both these conditions are also handled in `find_dep_version_match`, so just call `self.find_dep_version_match(dependency)` here?

Also, it's probably better to make `find_dep_version_match` return the picked version, rather than updating `dependency` in place, so:

```python
dependency['version'] = self.find_dep_version_match(dependency)

Reply:
done